wallets/react-wallet-v2/src/utils/WalletConnectUtil.ts

```diff
@@ -15,3 +15,39 @@ export async function createSignClient(relayerRegionURL: string) {
     }
   })
 }
+
+export async function updateSignClientChainId(chainId: string, address: string) {
+  const sessions = signClient.session.getAll()
+  for (const session of sessions) {
+    const fullChainName = getFullChainName(chainId)
+    const chain = fullChainName.substring(0, fullChainName.indexOf(':'))
+    const accounts = session.namespaces[chain] ? session.namespaces[chain].accounts : []
+    const new_account = `${fullChainName}:${address}`
+
+    // ensures no dup namespace accounts get added
+    if (!accounts.includes(new_account)) {
+      accounts.push(new_account)
+    }
+
+    const newNamespace = {
+      [chain]: {
+        accounts: accounts,
+        methods: [
+          'eth_sendTransaction',
+          'eth_signTransaction',
+          'eth_sign',
+          'personal_sign',
+          'eth_signTypedData'
+        ],
+        events: ['chainChanged', 'accountsChanged']
+      }
+    }
+
+    // console.log(session.namespaces ? { ...session.namespaces, ...newNamespace } : newNamespace)
+
+    signClient.update({
```

Comment:
I'm a bit confused about the intentions of this helper fn. `client.update()` is specifically to expand on the originally settled namespaces for the session, whereas "changing" chains/accounts is a superficial, UI-level event that won't alter the namespaces. I believe we need to use `client.emit()` for this, to simply signal this event to the peer client (i.e. the dapp in this case) so it can in turn update UI/own state. 

I think what we want to happen is very similar to what the sample wallet client is doing here in the UniversalProvider tests (see the helper above for the `accountsChanged` equivalent), where `client.emit(...)` is called when the chainId is updated:
https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/providers/universal-provider/test/shared/WalletClient.ts#L111

cc @ganchoradkov